docs/synapse/userguides/storm_adv_control.rstorm

```diff
@@ -115,27 +166,112 @@ You can print the value of a variable, to check its value at a given point in yo
   $asn=:asn
   $lib.print($asn)
 
-You can also print values associated with the node(s) in the current working set, using the various methods associated with the ``$node`` Storm type. (See :ref:`storm-adv-methods` for a user-focused introduction to methods, or :ref:`stormprims-storm-node-f527` in the detailed Storm Libraries / Storm Types documentation for a more technical discussion.)
+You can also print values associated with the node(s) in the current working set, using the various methods
+associated with the ``$node`` Storm type. (See :ref:`storm-adv-methods` for a user-focused introduction to methods,
+or :ref:`stormprims-storm-node-f527` in the detailed Storm Libraries / Storm Types documentation for a more
+technical discussion.)
 
 ::
   
   $lib.print($node.ndef())
 
 
-See the :ref:`storm-adv-example` below for an illustration of some of these effects.
-
+.. _storm-adv-control-flow:
 
 Control Flow Operations
 -----------------------
 
+.. TIP::
+  
+  The examples below are Storm excerpts used to illustrate specific concepts, but do not represent
+  complete Storm queries / Storm code.
+
+.. _flow-init:
+
+Init Block
+++++++++++
+
+An **init block** allows you to execute the specified Storm **once** at the beginning of your overall
+Storm query. This allows you to use Storm to perform a set of initial operations "outside" of the normal
+Storm pipeline; without the init block, the code would be executed once for **each** node passing through
+the Storm query.
+
+See also :ref:`flow-fini`.
+
+**Syntax:**
+
+::
+  
+  init { <storm> }
+
+
+**Example:**
+
+You want to use an init block to initialize a set of variables that will be used later in the Storm query.
+Initializing the variables to default values can:
+
+- Explicitly set a variable value up front.
+- Specify default values for variables in the event they are **not** set during subsequent execution
+  (e.g., due to a missing node, property, or tag that the variable depends on).
+- Initialize variables that will be modified during execution (e.g., lists, sets, tallies, or other 'count'
+  values you expect to change or increment).
+
+::
+  
+  init {
+  
+      $url=https://www.example.com/my_data/
+      $threatname=''
+      $fqdns=$lib.set()
+      $fqdn_count=0
+  }
+
+
+.. _flow-fini:
+
+Fini Block
+++++++++++
+
+A **fini block** allows you to execute the specified Storm **once** at the end of your overall Storm
```

Comment:
So a fini block executes once after all nodes in the pipeline have passed through, not at the end of the overall Storm query, so if you had a query like this for some reason:
```
cli> storm
for $x in $lib.range(3) {
  inet:fqdn=foo.com inet:fqdn=bar.com
  fini { $lib.print($x) }
} | spin

Executing query at 2022/10/27 22:48:30.529
0
1
2
```
Actually ends up executing that fini block once for each iteration of the loop, after all the lifted nodes have passed through.

Reply:
Not sure how much unusual usage of stuff like this we want to go into, but there are advanced cases where it is useful

Reply:
We can also have multiple fini blocks... which is probably an overcomplicated thing to discuss here.

Reply:
Reworded to attempt to clarify. Better?